app/javascript/mastodon/api_types/statuses.ts

```diff
@@ -13,6 +13,12 @@ export type StatusVisibility =
   // | 'limited' // This is never exposed to the API (they become `private`)
   | 'direct';
 
+// See app/modals/status.rb
+export type StatusFederation =
+  | 'federation_full'
+  | 'federation_limited'
```

Comment:
`federation_limited` is not required for local only, but I threw it in there with the idea of using it to only federate with allow listed servers.